### PR TITLE
fix(j-s): Court Arrangements

### PR DESCRIPTION
**Pull request by gudjong**
# Court Arrangements

[Böggur í date input á þingfestingar skjá/fyrirkalls skjá](https://app.asana.com/0/1199153462262248/1207541281422622/f)

## What

- Fixes time update in court arrangements. When editing the time part of the court date all data was cleared.

## Why

- Verified bug.

## Screenshots / Gifs

https://github.com/island-is/island.is/assets/795382/8088e074-94c6-4fb2-a083-c3c054b84dd7

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Enhanced handling of court arrangements with refined date and location management.

- **Bug Fixes**
  - Improved conditional logic for setting `courtEndTime` in CourtRecord component.

- **Refactor**
  - Updated various components to use `arraignmentDate` instead of `courtDate` for better consistency.
  - Simplified the logic within multiple functions to handle the new `DateLog` type appropriately.

### CODE REVIEW

File: apps/judicial-system/web/src/utils/validate.ts [hunk 1]
```diff
@@ -333,26 +334,33 @@ export const isReceptionAndAssignmentStepValid = (
...
       validate([
         [workingCase.defenderEmail, ['email-format']],
         [workingCase.defenderPhoneNumber, ['phonenumber']],
         [
-          courtDate ?? workingCase.arraignmentDate?.date,
+          arraignmentDate
+            ? arraignmentDate.date
+            : workingCase.arraignmentDate?.date,
           ['empty', 'date-format'],
         ],
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/utils/validate.ts:365]: The usage is correct.

File: apps/judicial-system/web/src/utils/validate.ts [hunk 2]
```diff
@@ -414,16 +422,22 @@
...
+          : workingCase.arraignmentDate?.date,
         ['empty', 'date-format'],
       ],
-      [courtRoom ?? workingCase.arraignmentDate?.location, ['empty']],
+      [
+        arraignmentDate
+          ? arraignmentDate.location
+          : workingCase.arraignmentDate?.location,
+        ['empty'],
+      ],
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/utils/validate.ts:440]: The usage is correct.

File: apps/judicial-system/web/src/components/CourtArrangements/CourtArrangements.tsx [hunk 3]
```diff
@@ -41,64 +41,55 @@
...
+          ).hasSent,
       )
+
+      setCourtDateHasChanged(courtDateHasChanged)
     }
+
+    setCourtDate((previous) => ({
+      ...previous,
+      date: date ? formatDateForServer(date) : null,
+    }))
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/components/CourtArrangements/CourtArrangements.tsx:85]: I want true boolean here.

File: apps/judicial-system/web/src/components/CourtArrangements/CourtArrangements.tsx [hunk 4]
```diff
@@ -41,64 +41,55 @@
   const { setAndSendCaseToServer } = useCase()
   const [original, setOriginal] =
     useState<[DateLogKey, DateLog | undefined | null]>()
-  const [courtDate, setCourtDate] = useState<DateLog | null>()
+  const [courtDate, setCourtDate] = useState<DateLog>({})
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/components/CourtArrangements/CourtArrangements.tsx:44]: No it does not.

File: apps/judicial-system/web/src/utils/validate.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/utils/validate.ts:440]: The usage is as intended. The argument passed into the method is of type DateLog and although it does not have the same name as the method parameter it is correct.

File: apps/judicial-system/web/src/components/CourtArrangements/CourtArrangements.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/components/CourtArrangements/CourtArrangements.tsx:44]: The type used here comes from a generated schema file:
```
export type DateLog = {
  __typename?: 'DateLog'
  date?: Maybe<Scalars['String']>
  location?: Maybe<Scalars['String']>
}
```

File: apps/judicial-system/web/src/utils/validate.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/utils/validate.ts:440]: It is.

File: apps/judicial-system/web/src/routes/Court/RestrictionCase/HearingArrangements/HearingArrangements.tsx [hunk 5]
```diff
@@ -173,7 +173,7 @@ export const HearingArrangements = () => {
             <CourtArrangements
               handleCourtDateChange={handleCourtDateChange}
               handleCourtRoomChange={handleCourtRoomChange}
-              courtDate={courtDate}
+              courtDate={workingCase.arraignmentDate}
```
Comment (unakb) [inline apps/judicial-system/web/src/routes/Court/RestrictionCase/HearingArrangements/HearingArrangements.tsx:176]: Please reconsider this change, it breaks the functionality of [using the requestedCourtDate the first time court user lands on the page](https://github.com/island-is/island.is/blob/1a09eda74a257821fa37c19ce8f449fb4e53c366/apps/judicial-system/web/src/routes/Court/RestrictionCase/HearingArrangements/HearingArrangements.tsx#L62).

**Review by unakb — CHANGES_REQUESTED** [review]
There's a change that accidentally removes the feature of displaying the requested court date when the user first lands on this hearing arrangements page.

File: apps/judicial-system/web/src/routes/Court/RestrictionCase/HearingArrangements/HearingArrangements.tsx (hunk 5 above)
Comment (gudjong) [reply apps/judicial-system/web/src/routes/Court/RestrictionCase/HearingArrangements/HearingArrangements.tsx:174]: Goot catch. I have fixed this on the hearing arrangement pages for both restriction cases and investigation cases.

**Review by unakb — APPROVED** [review]
Nice 😊

**Pull request merged**